### DISCUSSION TITLE
decimalSeparator and/or thousandsSeparator are messing things up

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#4819] [Help]
Not sure if this is a bug. I am still too much of a greenhorn to file it a such.

I have an integer field in the database. As it is supposed to store money values it contains cent values instead of euros. In `afterStateHydrated` the field's value will be divided by 100 to convert it to decimal.

**Given:** a field value of `123456` which would be `1234.56` with the default behaviour.
**Desired result:** `1.234,56`

So this works fine:
```
TextInput::make('price')
	->mask(fn (TextInput\Mask $mask) => $mask
		->numeric()
		->decimalPlaces(2)
	)
	// convert integer from database to decimal
	->afterStateHydrated(fn ($component, $state): TextInput => $component->state(round($state / 100, 2)))
```

Result: `1234.56`

This does not work:

```
TextInput::make('price')
	->mask(fn (TextInput\Mask $mask) => $mask
		->numeric()
		->decimalPlaces(2)
		->decimalSeparator(',')
		->thousandsSeparator('.')
	)
	// convert integer from database to decimal
	->afterStateHydrated(fn ($component, $state): TextInput => $component->state(round($state / 100, 2)))
```

Result: `123.456`

Am I missing something or is this actually a bug?

**Comment by danharrin** [comment]
Mask wont work with afterStateHydrated(), you'd need to use one or the other. We're going to replace the masking library in v3 so it's less buggy

**Reply by OvalMedia** [reply]
Tried `mutateFormDataBeforeFill` but the result is the same.

**Comment by fabioonet** [comment]
Hello, that's way worked for me:

```
Forms\Components\TextInput::make('price')
    ->mask(
        fn (TextInput\Mask $mask) => $mask
            ->patternBlocks([
                'money' => fn (Mask $mask) => $mask
                ->numeric()
                ->decimalPlaces(2)
                ->decimalSeparator(',')
                ->thousandsSeparator('.')
                ->mapToDecimalSeparator(['.'])
                ->padFractionalZeros()
                ->normalizeZeros(true)
        ])
        ->pattern('R$ money')
    )
    ->afterStateHydrated(function (TextInput $component, $state) {
        $component->state(str_replace(".",",",$state));
    })
,
```

I hope it helps
